bench/run_bench.py

```
@@ -0,0 +1,338 @@
+        u64 = (ctypes.c_uint64 * 40).from_buffer_copy(buf.raw[16:336])
+        return u64[29], u64[30]  # ri_instructions, ri_cycles
+
+    def counters_available():
+        return True
+
+else:
+    _PERF = shutil.which("perf")
+
+    def _perf_counts(pid, seconds):
+        """Sample instructions+cycles for `pid` over `seconds` via perf stat."""
+        out = subprocess.run(
```

Ruff (0.15.21)

[error] `subprocess` call: check for execution of untrusted input

(S603)